### ISSUE TITLE
[Desktop] MacOs, crashes when accessing U2F

### ISSUE THREAD

**Issue by nashbridges** [brave/brave-browser#10930]
## Description 

Apparently, Brave crashes when accessing U2F related code. 

## Steps to Reproduce

I have both github.com and facebook.com set up to require my hardware key (Yubikey 5c nano) as a 2FA.

#### Github

   1. when being signed out
   2. visit https://github.com/login
   3. a second later Brave crashes

#### Facebook

   1. when being signed out
   2. visit https://facebook.com
   3. enter login and password
   4. a screen "Touch your key" is shown
   5. a second later Brave crashes

## Actual result:

Brave crashes.

## Expected result:

Should allow to use my Yubikey.

## Reproduces how often: 

Every time.

## Brave version (brave://version info)

Brave | 1.11.101 Chromium: 84.0.4147.89 (Розробка) (64-розрядна версія)
-- | --
Редакція | 19abfe7bcba9318a0b2a6bc6634a67fc834aa592-refs/branch-heads/4147@{#852}
ОС | macOS Версія 10.12.6 (складення 16G2136)
JavaScript | V8 8.4.371.19
Flash | (Вимкнено)
Агент користувача | Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/84.0.4147.89 Safari/537.36
Командний рядок | /Applications/Brave Browser.app/Contents/MacOS/Brave Browser -psn_0_475252 --enable-dom-distiller --disable-domain-reliability --no-pings --extension-content-verification=enforce_strict --extensions-install-verification=enforce --sync-url=https://no-thanks.invalid --enable-features=AutoupgradeMixedContent,PassiveMixedContentWarning,PasswordImport,WebUIDarkMode,MixedContentSiteSetting,DnsOverHttps --disable-features=AutofillServerCommunication,AutofillEnableAccountWalletStorage,AllowPopupsDuringPageUnload,TextFragmentAnchor,LookalikeUrlNavigationSuggestionsUI,UseButtonTranslateBubbleUI,VideoPlaybackQuality,NotificationTriggers,SmsReceiver,TabHoverCards --flag-switches-begin --load-media-router-component-extension=1 --flag-switches-end --enable-audio-service-sandbox --disable-sync
Виконуваний шлях | /Applications/Brave Browser.app/Contents/MacOS/Brave Browser
Шлях до профілю | /Users/nash/Library/Application Support/BraveSoftware/Brave-Browser/Default

## Version/Channel Information:

- Can you reproduce this issue with the current release? n/a
- Can you reproduce this issue with the beta channel? n/a
- Can you reproduce this issue with the nightly channel? n/a

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? no
- Does the issue resolve itself when disabling Brave Rewards? no
- Is the issue reproducible on the latest version of Chrome? no (works fine on 84.0.4147.89)

The issue is not reproducible in incognito mode.
The issue *is* reproducible on a new clean profile (no extensions).
It does not matter do I have the key inserted in a USB port or not.

## Miscellaneous Information:

[log.txt](https://github.com/brave/brave-browser/files/4982145/log.txt)

**Comment by rebron** [comment]
cc: @jumde

**Comment by jumde** [comment]
@nashbridges - Thanks for the report. We are not able to repro this on: MacOS Catalina, Mojave or High-Sierra. @kjozwiak - is trying the steps on Sierra. 

We are also looking into the stack-trace to see if we can figure out the root cause. Will post updates as we have them. Thanks again!

**Comment by kjozwiak** [comment]
@nashbridges mind taking a look at `brave://crashes` and reporting the `Crash Report ID`? Those would also help figure out why you're seeing the crash as those crashes will already be symbolized. You'll first need to send the report if you have `Automatically sends crash reports to Brave` disabled via `brave://settings/privacy`.

**Comment by nashbridges** [comment]
@kjozwiak 6d480000-45d5-d704-0000-000000000000

**Comment by nashbridges** [comment]
> * Is the issue reproducible on the latest version of Chrome? no (works fine on 84.0.4147.89)

I've just updated Chrome to 84.0.4147.105, and now it also crashes.
So probably it's an upstream bug with Chromium + Sierra.

Not sure if you guys can help with this. I do understand that my OS is way too old.
Feel free to close it.

**Comment by kjozwiak** [comment]
@jumde looks like the following crash stack but but 100% sure if that's related to the U2F library.

```
[ 00 ] (anonymous namespace)::ZombieObjectCrash(objc_object*, objc_selector*, objc_selector*)
[ 01 ] -[CrZombie forwardingTargetForSelector:]
[ 02 ] ___forwarding___
[ 03 ] __forwarding_prep_0___
[ 04 ] (anonymous namespace)::AutoreleasePoolPage::pop(void*)
[ 05 ] _CFAutoreleasePoolPop
[ 06 ] -[NSAutoreleasePool drain]
[ 07 ] base::mac::CallWithEHFrame(void () block_pointer)
[ 08 ] base::MessagePumpCFRunLoopBase::RunWorkSource(void*)
[ 09 ] __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__
[ 10 ] __CFRunLoopDoSources0
[ 11 ] __CFRunLoopRun
[ 12 ] CFRunLoopRunSpecific
[ 13 ] RunCurrentEventLoopInMode
[ 14 ] ReceiveNextEventCommon
[ 15 ] _BlockUntilNextEventMatchingListInModeWithFilter
[ 16 ] _DPSNextEvent
[ 17 ] -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:]
[ 18 ] __71-[BrowserCrApplication nextEventMatchingMask:untilDate:inMode:dequeue:]_block_invoke
[ 19 ] base::mac::CallWithEHFrame(void () block_pointer)
[ 20 ] -[BrowserCrApplication nextEventMatchingMask:untilDate:inMode:dequeue:]
[ 21 ] -[NSApplication run]
[ 22 ] base::MessagePumpNSApplication::DoRun(base::MessagePump::Delegate*)
[ 23 ] base::MessagePumpCFRunLoopBase::Run(base::MessagePump::Delegate*)
[ 24 ] base::sequence_manager::internal::ThreadControllerWithMessagePumpImpl::Run(bool, base::TimeDelta)
[ 25 ] base::RunLoop::Run()
[ 26 ] ChromeBrowserMainParts::MainMessageLoopRun(int*)
[ 27 ] content::BrowserMainLoop::RunMainMessageLoopParts()
[ 28 ] content::BrowserMainRunnerImpl::Run()
[ 29 ] content::BrowserMain(content::MainFunctionParams const&)
[ 30 ] content::ContentMainRunnerImpl::RunServiceManager(content::MainFunctionParams&, bool)
[ 31 ] content::ContentMainRunnerImpl::Run(bool)
[ 32 ] service_manager::Main(service_manager::MainParams const&)
[ 33 ] content::ContentMain(content::ContentMainParams const&)
[ 34 ] ChromeMain
[ 35 ] main
[ 36 ] start
```

**Comment by kjozwiak** [comment]
> @kjozwiak - is trying the steps on Sierra. 

@jumde can't seem to reproduce any crashes on my `Sierra` VM using `1.11.104 CR: 84.0.4147.105`. Example:

![SierraYubi](https://user-images.githubusercontent.com/2602313/89804980-ecf33200-db02-11ea-8628-e86fa926ea3b.gif)

@nashbridges let me know if I'm doing anything incorrectly re: the above issues but sounds like you're experiencing it on other browsers as well.

**Comment by nashbridges** [comment]
@kjozwiak 

> let me know if I'm doing anything incorrectly re: the above issues

I don't even have a chance to enter my credentials on github.com/login, Brave crashes a moment later after I visit the page.

> sounds like you're experiencing it on other browsers as well

It crashes with Brave 1.11.104 and Chrome 84.0.4147.125.
It does not crash with Firefox 79.

As I said, this is probably an upstream bug in Chromium, because I didn't experience it previously with Brave. Let's hope they will fix it soon.

**Comment by nashbridges** [comment]
I confirm the issue is fixed in Chrome 85.0.4183.102

**Comment by jumde** [comment]
Thanks for the update @nashbridges. Closing. Feel free to re-open/file a new issue if you still see this issue.
